x/ccv/provider/keeper/proposal.go

```diff
@@ -332,11 +335,18 @@ func (k Keeper) BeginBlockInit(ctx sdk.Context) {
 	propsToExecute := k.ConsumerAdditionPropsToExecute(ctx)
 
 	for _, prop := range propsToExecute {
-		p := prop
-		err := k.CreateConsumerClient(ctx, &p)
+		// create consumer client in a cached context to handle errors
```

Comment:
Is there any significant performance impact when a new cached ctx is created for every prop inside the loop?

Is there a problem with using
```golang
cacheCtx, writeCacheFunc = ctx.CacheContext()
for _, props := range propsToExecute {
    	err := k.CreateConsumerClient(cacheCtx, prop)
        if err != nil {
            continue
        }
        writeCacheFunc()
}
```

Reply:
I don't see any, good catch

Reply:
Re your snippet @MSalopek I'm not sure if I understand the ctx caching exactly, but isn't it a problem if you `continue` and then afterwards `writeCacheFunc()` in a later iteration? The bogus data from the iteration when you `continued` will still be in the cacheCtx and then it will be written.

Reply:
I'm not sure how this works, but what you state might be why there was a separate cache context for each iteration.

Good catch!

Reply:
That would just undo all props, not just the one that failed